### ISSUE TITLE
Scroll up/down action?

### ISSUE THREAD

**Issue by SheetLightning** [stefansundin/powermate-linux#16]
I'm trying to figure out how to do a scroll up/down action instead of volume up/down action? There seems to be no indixation as to what paremeters to use in the config file?

**Comment by stefansundin** [comment]
Hi there,

I have never tried this specific use case, but I was googling a bit on xdotool, and I think you can accomplish this with `xdotool key Page_Up` and `xdotool key Page_Down`. I think page up/down may scroll too much, so maybe `Up` and `Down` work better. You may be able to simulate scroll events with `xdotool click 4` and `xdotool click 5`.

I hope you get it working to your satisfaction.

**Comment by SheetLightning** [comment]
Thank you. I tried "xdotool key Up" and "xdotool key Down" and this works fine. You were quite correct that although xdotool key Page_Up / Page_Down both work as well, the scrolling action was rather too fast.

I am still a little confused as regards using the Powermate on Linux, as the Linux kernel (I am using Linux Mint 19.2, which is based on Ubuntu) seems to have a built-in driver (as shown by lsmod) but I can't find any documentation as to how to use it or whether additional tools such as this one are actually required to make the Powermate work. In any case I am glad that I found your tool while researching the subject so can I say thank you for making it available. :-)

Is it possible to combine actions, for example to perform scroll up/down while the button is rotated but volume up/down while the button is pressed and rotated?

**Comment by stefansundin** [comment]
That functionality should be possible to add to this program, but it is not currently supported.

If you'd like to experiment yourself, you should be able to add it by adding a new `if/else` statement in this part of the code:
https://github.com/stefansundin/powermate-linux/blob/28bd3e1cf736f1dc2c230f312232e1c821c3181f/main.c#L199-L213

You'd basically want to check `if (knob_depressed) {` and you can just run a hard-coded `exec_command()` for now. Once you get it working, you can look at the other code to add a new option to the toml file.

**Comment by SheetLightning** [comment]
Thank you for the suggestion. I had a play with main.c and I have it working. I have added a couple of command parameters to the powermate.toml file to provide configurable rotate actions when the button is pressed down. Working around the short and long button press and the timeout was the trickiest part to figure out but I think I have managed it acceptably and in keeping with your style. I attach my main.c and powermate.toml file for you to compare.

[powermate-linux-jch-contrib.zip](https://github.com/stefansundin/powermate-linux/files/3975737/powermate-linux-jch-contrib.zip)
